packages/cli/src/commands/commerce/catalog/create.ts

```diff
@@ -66,74 +70,134 @@ export default class CatalogCreate extends Command {
     // TODO: Add edit catalog privilege. https://docs.coveo.com/en/2956/coveo-for-commerce/index-commerce-catalog-content-with-the-stream-api#required-privileges
   )
   public async run() {
-    this.printWarningMessage();
-    const client = await new AuthenticatedClient().getClient();
-    const {fields, objectTypes} = await this.parseDocuments(client);
+    const {flags} = await this.parse(CatalogCreate);
+    const authenticatedClient = new AuthenticatedClient();
+    const client = await authenticatedClient.getClient();
+    const configuration = authenticatedClient.cfg.get();
+    const catalogConfigurationModel = await this.generateCatalogConfiguration();
 
-    await this.promptQuestions(fields, objectTypes);
     await this.ensureCatalogValidity();
 
     // Destructive changes starting from here
-    const configuration = await this.createCatalogConfiguration(client);
-    await this.createCatalog(client, configuration);
+    const {productSourceId, catalogSourceId} = await this.createSources(
+      client,
+      catalogConfigurationModel
+    );
+    CliUx.ux.action.start('Creating catalog');
+    const {id: catalogConfigurationId} = await this.createCatalogConfiguration(
+      client,
+      catalogConfigurationModel
+    );
+    const catalog = await this.createCatalog(
+      client,
+      catalogConfigurationId,
+      productSourceId,
+      catalogSourceId
+    );
+    await this.mapStandardFields(
+      productSourceId,
+      catalogConfigurationId,
+      configuration
+    );
+    // TODO: CDX-1022: make id fields facetable
+    if (flags.output) {
+      CliUx.ux.styledJSON(catalog);
+    }
   }
 
   @Trackable()
   public async catch(err?: Error & {exitCode?: number}) {
     throw err;
   }
 
-  private async parseDocuments(client: PlatformClient) {
+  protected async finally(err: Error | undefined) {
+    CliUx.ux.action.stop(err ? red.bold('!') : green('✔'));
+  }
+
+  private async generateCatalogConfiguration(): Promise<PartialCatalogConfigurationModel> {
     const {flags} = await this.parse(CatalogCreate);
-    const analyser = new FieldAnalyser(client);
-    const docBuilders: DocumentBuilder[] = [];
-    const objectTypeValues: Set<MetadataValue> = new Set();
-    const callback = async (docBuilder: DocumentBuilder, docPath: PathLike) => {
-      const {metadata, uri} = docBuilder.build();
-      if (metadata?.objecttype === undefined) {
-        CliUx.ux.warn(
-          `missing ${bold('objecttype')} metadata on item ${uri} (${docPath})`
-        );
-      } else {
-        objectTypeValues.add(metadata.objecttype);
-      }
+    try {
+      CliUx.ux.action.start('Generating catalog configuration from data');
+      return getCatalogPartialConfiguration(flags.dataFiles);
+    } catch (error) {
```

Comment:
Is handling this as an thrown error the correct thing to do here ?

As opposed to just a "normal" return type from `getCatalogPartialConfiguration` function ?

I'm asking this because, naturally, this does not really seem like an error to me, it's just that we could not the the automatic job. 

So it seems like a denaturation of what an error should be.

Reply:
I am using a try catch because this is a very early implementation of the automatic catalog detection feature and a lot of things could go wrong here. 
Also, the detection currently does not work for every catalog type (e.g. catalogs with availabilities).

So to keep it simple, I just catch any error and redirect to the interactive mode.